server/app/views/admin/apikeys/ApiKeyIndexView.java

```diff
@@ -0,0 +1,156 @@
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.inject.Inject;
+import j2html.tags.ContainerTag;
+import java.util.function.Function;
+import models.ApiKey;
+import play.mvc.Http;
+import play.twirl.api.Content;
+import services.DateConverter;
+import services.PaginationResult;
+import views.BaseHtmlView;
+import views.HtmlBundle;
+import views.admin.AdminLayout;
+import views.components.LinkElement;
+import views.style.ReferenceClasses;
+import views.style.Styles;
+
+/** Renders a page that lists the system's {@link models.ApiKey}s. */
+public final class ApiKeyIndexView extends BaseHtmlView {
+  private final AdminLayout layout;
+  private final DateConverter dateConverter;
+  private final Slugify slugifier = new Slugify();
+
+  @Inject
+  public ApiKeyIndexView(AdminLayout layout, DateConverter dateConverter) {
+    this.layout = checkNotNull(layout);
+    this.dateConverter = checkNotNull(dateConverter);
+  }
+
+  public Content render(
+      Http.Request request,
+      PaginationResult<ApiKey> apiKeyPaginationResult,
+      ImmutableSet<String> allProgramNames) {
+    String title = "API Keys";
+    ContainerTag headerDiv =
+        div()
+            .withClasses(Styles.FLEX, Styles.PLACE_CONTENT_BETWEEN, Styles.MY_8)
+            .with(
+                h1(title).withClasses(Styles.MY_4),
+                new LinkElement()
+                    .setHref(controllers.admin.routes.AdminApiKeysController.newOne().url())
+                    .setId("new-api-key-button")
+                    .setText("New API Key")
+                    .asButton());
+
+    ContainerTag contentDiv = div().withClasses(Styles.PX_20).with(headerDiv);
+
+    for (ApiKey apiKey : apiKeyPaginationResult.getPageContents()) {
+      contentDiv.with(renderApiKey(request, apiKey, buildProgramSlugToName(allProgramNames)));
+    }
+
+    HtmlBundle htmlBundle = layout.getBundle().setTitle(title).addMainContent(contentDiv);
+    return layout.renderCentered(htmlBundle);
+  }
+
+  private ContainerTag renderApiKey(
+      Http.Request request, ApiKey apiKey, ImmutableMap<String, String> programSlugToName) {
+    ContainerTag statsDiv =
+        div()
+            .with(
+                p("Created " + dateConverter.formatRfc1123(apiKey.getCreateTime())),
+                p("Created by " + apiKey.getCreatedBy()),
+                p(
+                    apiKey
+                        .getLastCallIpAddress()
+                        .map(ip -> "Last used by " + ip)
+                        .orElse("Last used by N/A")),
+                p("Call count: " + apiKey.getCallCount()))
+            .withClasses(Styles.TEXT_XS);
+
+    ContainerTag linksDiv = div().withClasses(Styles.FLEX);
+
+    if (apiKey.isRetired()) {
+      statsDiv.with(
+          p("Retired " + dateConverter.formatRfc1123(apiKey.getRetiredTime().get())),
+          p("Retired by " + apiKey.getRetiredBy().get()));
```

Comment:
I'm hesitant to do that since we now understand that email is not guaranteed to be uniquely identified with the account.

My thinking with plugging in the authority ID here was that if there was some suspicious behavior the admin could send the authority ID to whoever is in charge of their identity provider who could look up the authoritative user account.

How unfit is it? Like a lengthy guid?

Reply:
From a UX perspective: The product doesn't have Authority ID as a User visible concept, so we should consider what the Admin experience is having a value who's format has no context is.

Data: so far it's been the users email and a URL of the auth provider, which we format as "sub: XXX iss: XXXX" , so not short, but not god awful or gibberish per se, but it can be anything.

This seems like the same problem an Admin has when they're managing TIs, (eg finding them to remove later) what do we do there?

Reply:
We talked out-of-band and I'm going to remove retiredBy from the UI for now.